api/hasher/hasher_test.go

```diff
@@ -1,12 +1,13 @@
 // Copyright 2019 The Kubernetes Authors.
 // SPDX-License-Identifier: Apache-2.0
 
-package hasher_test
+package hasher
```

Comment:
It appears that all the tests follow the same pattern
- make a yaml object
- compute its hash
- compare to some constant value encoded in the test.

So maybe just keep this package as `hasher_test`, and only call the _public_ method `HashRNode` in all cases, rather than calling the private, individually typed methods (`encodeConfigMap`, `encodeSecretHash`, etc).

Or is the latter approach somehow a better test and i'm not seeing it?

regardless, there should be at least a few tests calling the public method.

Reply:
These tests are from old hash implementations. I am happy to change it to use public methods.

Reply:
But `encodeXXX` methods are private and I think tests for them are necessary. So I will keep the `hasher` package name.